src/components/__tests__/icon-tests.js

```diff
@@ -0,0 +1,141 @@
+jest.dontMock('../icon');
+jest.dontMock('jquery');
+jest.dontMock('lodash');
+
+var $ = require('jquery');
+var _ = require('lodash');
+
+describe('Icon', function() {
+  var Icon, icon, React, TestUtils;
+
+  beforeEach(function() {
+    React = require('react/addons');
+    TestUtils = React.addons.TestUtils;
+    Icon = require('../icon');
+    icon = <Icon />;
+  });
+
+  it('is a Icon element', function() {
+    expect(TestUtils.isElementOfType(icon, Icon)).toEqual(true);
+  });
+
+  it('is defined', function() {
+    expect(Icon).toBeDefined();
+  });
+
+  it('defaults type to default', function() {
+    var elem = TestUtils.renderIntoDocument(icon);
+    expect(elem.props.type).toEqual('default');
+  });
+
+  it('has the icon class', function() {
+    var elem = TestUtils.renderIntoDocument(icon);
+    var node = elem.getDOMNode();
+    expect($(node).hasClass('icon')).toEqual(true);
+  });
+
+  describe('type', function() {
+    it('no icon type class is set by default', function() {
+      var elem = TestUtils.renderIntoDocument(icon);
+      var node = elem.getDOMNode();
+      var iconClasses = [
+        'icon-default',
+        'icon-back',
+        'icon-bars',
+        'icon-caret',
+        'icon-check',
+        'icon-close',
+        'icon-code',
+        'icon-compose',
+        'icon-download',
+        'icon-edit',
+        'icon-forward',
+        'icon-gear',
+        'icon-home',
+        'icon-info',
+        'icon-list',
+        'icon-more-vertical',
+        'icon-more',
+        'icon-pages',
+        'icon-pause',
+        'icon-person',
+        'icon-play',
+        'icon-plus',
+        'icon-refresh',
+        'icon-search',
+        'icon-share',
+        'icon-sound',
+        'icon-sound2',
+        'icon-sound3',
+        'icon-sound4',
+        'icon-star-filled',
+        'icon-star',
+        'icon-stop',
+        'icon-trash',
+        'icon-up-nav',
+        'icon-up',
+        'icon-right-nav',
+        'icon-right',
+        'icon-down-nav',
+        'icon-down',
+        'icon-left-nav',
+        'icon-left'
+      ];
+      _.each(iconClasses, function(iconClass) {
+        expect($(node).hasClass('icon-default')).toEqual(false);
+      });
```

Comment:
I think maybe what you were trying to do was create a `node` instance of each `iconClass`? If so, you'd want to check that `icon-default` is _not_ set and that `iconClass` _is set_ I think.